Add tvOS support

sergiou87 opened the pull request.

bang590 commented:
不支持tvos

sergiou87 commented:
Hi @bang590 !

Sorry, but what's the reason for that? I tried it and worked perfectly. Is there a known issue or limitation using JSPatch on tvOS?

bang590 commented:
```
- ERROR | [tvOS] [JSPatch/Extensions] xcodebuild: Returned an unsuccessful exit code.
- ERROR | [tvOS] [JSPatch/Extensions] xcodebuild:  JSPatch/Extensions/JPSpecialInit.h:21:4: error: 'UIWebView' is unavailable: not available on tvOS
```
Have you run the ` pod spec lint JSPatch.podspec` ? Extensions didn't support tvOS, and it would be better if there is a tvOS demo project.

sergiou87 commented:
Oh! I see... so not _everything_ is supported on tvOS. Sorry about that!

I'll try to come up with a more exhaustive Pull Request 😄

sergiou87 commented:
Hey @bang590 !

Please, take a look again, hopefully it's fine now 😄 

I've fixed the build problems on tvOS and added a demo for Apple TV. Let me know if there is anything else I should do 😄
